### PR TITLE
Filecoin support

### LINKED ISSUE #20: Support Filecoin / Elrond

### PR DESCRIPTION
**Pull request by catenocrypt**
Fixes #20

### CODE REVIEW

File: filecoin.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		return TypeBls
+	default:
+		return TypeInvalid
+	}
+}
+
+// Returns the payload size (excluding any prefixes) of an address type.
+// If the payload size is undefined/variable (e.g. ID)
+// or the type is unknown, it returns zero.
+func payloadSize(typ Type) byte {
```
Comment (hewigovens) [inline filecoin.go:68]:
```suggestion
func payloadSize(t Type) byte {
```

File: filecoin.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+func isValidString(input string) error {
+	if len(input) < 3 {
+		return errors.New("Too short")
+	}
+	// Only main net addresses supported.
+	if input[0] != PREFIX {
+		return errors.New("Invalid network")
+	}
+	// Get address type.
+	typ := parseType(input[1])
```
Comment (hewigovens) [inline filecoin.go:133]:
```suggestion
	t := parseType(input[1])
```
golang style :)

File: filecoin.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	bytes = append(bytes, decoded[:payloadSize]...)
+	return bytes, nil
+}
+
+// FilecoinEncodeToString converts the input byte array to a string representation of the Solana address.
+func FilecoinEncodeToString(data []byte) (string, error) {
+	if len(data) < 2 {
+		return "", errors.New("Data too short")
+	}
+	var typ Type = getType(data[0])
```
Comment (hewigovens) [inline filecoin.go:180]:
```suggestion
	var t Type = getType(data[0])
```

**Comment by catenocrypt** [comment]
ensdomains FIL test are included here as well
https://github.com/ensdomains/address-encoder/blob/a938409270d292a0714a34b6671caab5baeecc6d/src/__tests__/index.test.ts#L541

**Review by hewigovens — APPROVED** [review]
nice work

**Pull request merged**